src/debugger_core.jl

```diff
@@ -46,15 +45,17 @@ function attempt_to_set_f_breakpoints!(bps)
 end
 
 function get_next_top_level_frame(state)
-    state.current_top_level_expression += 1
-
-    if state.current_top_level_expression > length(state.top_level_expressions)
-        return nothing
-    else
-        next_top_level = state.top_level_expressions[state.current_top_level_expression]
-        next_frame = JuliaInterpreter.prepare_thunk(next_top_level)
-        return next_frame
+    x = iterate(state.expr_splitter)
+    x === nothing && return nothing
+
+    mod, ex = x
```

Comment:
```suggestion
    (mod, ex), _ = x
```
since `x[2]` should (I think) be the state variable.

Reply:
thanks, I just missed what `iterate` returns.